lib/perl/Genome/InstrumentData/Command/Import/WorkFlow/FastqsToBam.pm

```diff
@@ -46,16 +60,76 @@ sub execute {
     my $verify_bam_ok = $self->_verify_bam;
     return if not $verify_bam_ok;
 
-    $self->debug_message('Fastqs to bam...done');
+    $self->status_message('Fastqs to bam...done');
+    return 1;
+}
+
+sub _unarchive_fastqs_if_necessary {
+    my $self = shift;
+    $self->status_message('Unarchive fastqs if necessary...');
+
+    my @new_fastq_paths;
+    for my $fastq_path ( $self->fastq_paths ) {
+        if ( $fastq_path !~ /\.gz$/ ) {
+            $self->status_message('Unarchive not necessary for '.$fastq_path);
+            push @new_fastq_paths, $fastq_path;
+            next;
+        }
+
+        $self->status_message('Unarchiving: %s', $fastq_path);
+        my $unarchived_fastq_path = $self->get_working_path_for_file_path($fastq_path);
+        $unarchived_fastq_path =~ s/\.gz//;
+        $self->status_message('To: %s', $unarchived_fastq_path);
+        my $extractor = Archive::Extract->new(archive => $fastq_path);
+        if ( not $extractor->extract(to => $unarchived_fastq_path) ) {
+            $self->error_message( $extractor->error ) if $extractor->error;
+            $self->error_message('Archive::Extract failed!');
+            return;
+        }
+
+        if ( not -s $unarchived_fastq_path ) {
+            $self->error_message('Unarchived fastq does not exist!');
+            return;
+        }
+        push @new_fastq_paths, $unarchived_fastq_path;
+        unlink $fastq_path;
+    }
+    $self->fastq_paths(\@new_fastq_paths);
+
+    $self->status_message('Unarchive fastqs if necessary...');
     return 1;
 }
 
+sub _get_fastq_read_counts {
+    my $self = shift;
+    $self->status_message('Getting fastq read count...');
+
+    my @line_counts;
+    for my $fastq_path ( $self->fastq_paths ) {
+        $self->status_message('Fastq: %s', $fastq_path);
+        open(my $FILE, $fastq_path);
+        1 while <$FILE>;
+        my $line_count = $.;
```

Comment:
[The perlfaq](http://perldoc.perl.org/perlfaq5.html#How-do-I-count-the-number-of-lines-in-a-file%3f) suggests that shelling out is the fastest option, but maybe one of those chunking algorithms would be suitable?